### PR TITLE
[MRG+1] Mep12 text labels and annotations

### PR DESCRIPTION
**Pull request by NelleV**
Here are some more migrations of examples to compatible sphinx-gallery docstring.

refs #7206

### CODE REVIEW

File: examples/text_labels_and_annotations/text_demo_fontdict.py [hunk 1]
```diff
@@ -1,5 +1,11 @@
 """
-Demo using fontdict to control style of text and labels.
+=======================================================
+Controlling style of text and labels using a dictionary
+=======================================================
+
+
```
Comment (QuLogic) [inline examples/text_labels_and_annotations/text_demo_fontdict.py]: Extra line.

File: examples/text_labels_and_annotations/text_demo_fontdict.py [hunk 2]
```diff
@@ -1,5 +1,11 @@
...
+This example shows how to share parameters across many text objects and labels
+by creating a dictionnaries of option passed across several functions.
```
Comment (QuLogic) [inline examples/text_labels_and_annotations/text_demo_fontdict.py]: dictionnaries of option -> dictionary of options

**Comment by NelleV** [comment]
Thanks for the review! I fixed the comments.

File: examples/text_labels_and_annotations/text_demo_fontdict.py [hunk 3]
```diff
@@ -1,5 +1,10 @@
 """
-Demo using fontdict to control style of text and labels.
+=======================================================
+Controlling style of text and labels using a dictionary
+=======================================================
+
+This example shows how to share parameters across many text objects and labels
+by creating a dictionnary of options passed across several functions.
```
Comment (QuLogic) [inline examples/text_labels_and_annotations/text_demo_fontdict.py]: dictionnary ->dictionary

Comment (NelleV) [inline examples/text_labels_and_annotations/text_demo_fontdict.py]: fixed again :)

File: examples/text_labels_and_annotations/autowrap_demo.py [hunk 4]
```diff
@@ -1,5 +1,11 @@
 """
-Auto-wrapping text demo.
+==================
+Auto-wrapping text
+==================
+
+Matplotlib can wrap text automatically, but if it's too long, the text will be
+displayed slightly outside of the boundaries of the axis anyways.
+
```
Comment (anntzer) [inline examples/text_labels_and_annotations/autowrap_demo.py]: A quick grep suggests that the common style in the codebase is an empty line after the closing quotes, not before.

Also applies below.

Comment (NelleV) [inline examples/text_labels_and_annotations/autowrap_demo.py]: We have common style code for documenting examples? And here I thought we didn't have documentation on examples :p

**Comment by NelleV** [comment]
I've added/removed blank lines.

**Comment by NelleV** [comment]
So… The documentation passes. The rest fails because it can't download freetype.

**Comment by dstansby** [comment]
Backported to `2.0.x` via. https://github.com/matplotlib/matplotlib/commit/605e57b4b1a7f01bf5b96e9ebf2496f7c3cbdc78

**Pull request merged**